src/main/java/ui/screens/CycleTimeScreen.java

```diff
@@ -24,30 +39,140 @@ protected void beforeVisualizationMount() {
+                if (cycleTime != null) {
+                    calendarUpdate();
+                }
+
+                //disable all days in the EndDate picker before the start date
+                endDate.setDayCellFactory(picker -> new DateCell() {
+                    @Override
+                    public void updateItem(LocalDate date, boolean empty) {
+                        super.updateItem(date, empty);
+                        setDisable(empty || date.isBefore(startDate.getValue().plusDays(1)));
+                    }
+                });
+            }
+        };
+        this.startDate.valueProperty().addListener(startDateChangeListener);
+
+        ChangeListener<LocalDate> endDateChangeListener = new ChangeListener<LocalDate>() {
+            @Override
+            public void changed(ObservableValue<? extends LocalDate> observableValue,
+                                LocalDate localDate, LocalDate t1) {
+                if (cycleTime != null) {
+                    calendarUpdate();
+                }
+
+                //disable all days in the StartDate picker after the end date
+                startDate.setDayCellFactory(picker -> new DateCell() {
+                    @Override
+                    public void updateItem(LocalDate date, boolean empty) {
+                        super.updateItem(date, empty);
+                        setDisable(empty || date.isAfter(endDate.getValue()));
+                    }
+                });
+            }
+        };
+        this.endDate.valueProperty().addListener(endDateChangeListener);
+
         sprint_combobox.setOnAction((e) -> {
             Sprint sprint = sprint_combobox.getValue();
             if (sprint == null) {
                 return;
             }
-            this.cycleTime.switchSprint(sprint);
+            CycleTime temp = cycleTime;
+            cycleTime = null; //prevent calendars from updating the actual service
+
+            this.startDate.setValue(DateUtil.toLocal(sprint.getEstimatedStart()));
+            this.endDate.setValue(DateUtil.toLocal(sprint.getEstimatedFinish()));
+
+            cycleTime = temp;
+            sprint_name.setText(sprint.getName());
+            updateCycleTimeSprint();
         });
+    }
+
+    /**
+     * Adds the switch sprint functionality to the sprint dropdown after the page is rendered.
+     */
+    @Override
+    protected void afterVisualizationMount() {
+
         sprint_combobox.getSelectionModel().selectLast();
-        this.cycleTime.switchSprint(sprint_combobox.getValue());
+
+        Sprint sprint = sprint_combobox.getValue();
+
+        CycleTime temp = cycleTime;
+        cycleTime = null; //prevent calendars from updating the actual service
+        this.startDate.setValue(DateUtil.toLocal(sprint.getEstimatedStart()));
+        this.endDate.setValue(DateUtil.toLocal(sprint.getEstimatedFinish()));
+        cycleTime = temp;
+
+        sprint_name.setText(sprint.getName());
+        updateCycleTimeSprint();
```

Comment:
We should likely only use dates for computing cycle time. Using sprints adds extra complexity (namely the temp CycleTime variable) since we only use the dates from the selected sprint.

Reply:
My initial thought is this: rather than having two separate methods (`switchDates` and `switchSprint`), we have a single one that accepts all parameters. From there we can either:
1. Keep the same method overrides in `CycleTimeService` (keep `switchSprint` and `switchDates` in CycleTimeService). We would have to push the logic up of determining which method to use.
2. Refactor `CycleTimeService` to also have a single `recalculate` implementation.